static/app/components/replays/replayController.tsx

```diff
@@ -15,117 +15,131 @@ import {formatTime} from './utils';
+      name="replay-timeline"
+      min={0}
+      max={duration}
+      value={Math.round(currentTime)}
+      onChange={value => setCurrentTime(value || 0)}
+      showLabel={false}
+    />
+  );
 }
 
-const ReplayControls = ({
-  currentTime,
-  duration,
-  isFullscreen,
-  isPlaying,
-  isSkippingInactive,
-  setCurrentTime,
-  setSpeed,
-  speed,
-  speedOptions,
-  toggleFullscreen,
-  togglePlayPause,
-  toggleSkipInactive,
-}: ControlsProps) => {
+function ReplayPlayPauseBar() {
+  const {currentTime, isPlaying, setCurrentTime, togglePlayPause} = useReplayContext();
+
   return (
-    <React.Fragment>
-      <TimelineRange
-        data-test-id="replay-timeline-range"
-        name="replay-timeline"
-        min={0}
-        max={duration}
-        value={Math.round(currentTime)}
-        onChange={value => setCurrentTime(value || 0)}
-        showLabel={false}
+    <ButtonBar merged>
+      <Button
+        data-test-id="replay-back-10s"
+        size="xsmall"
+        title={t('Go back 10 seconds')}
+        icon={<IconRefresh size="sm" />}
+        onClick={() => setCurrentTime(currentTime - 10 * SECOND)}
+        aria-label={t('Go back 10 seconds')}
+      />
+      <Button
+        data-test-id="replay-play-pause"
+        size="xsmall"
+        title={isPlaying ? t('Pause the Replay') : t('Play the Replay')}
+        icon={isPlaying ? <IconPause size="sm" /> : <IconPlay size="sm" />}
+        onClick={() => togglePlayPause(!isPlaying)}
+        aria-label={isPlaying ? t('Pause the Replay') : t('Play the Replay')}
       />
+      <Button
+        data-test-id="replay-forward-10s"
+        size="xsmall"
+        title={t('Go forward 10 seconds')}
+        icon={<IconClockwise size="sm" />}
+        onClick={() => setCurrentTime(currentTime + 10 * SECOND)}
+        aria-label={t('Go forward 10 seconds')}
+      />
+    </ButtonBar>
+  );
+}
 
-      <ButtonGrid>
-        <ButtonBar merged>
-          <Button
-            data-test-id="replay-back-10s"
-            size="xsmall"
-            title={t('Go back 10 seconds')}
-            icon={<IconRefresh size="sm" />}
-            onClick={() => setCurrentTime(currentTime - 10 * SECOND)}
-            aria-label={t('Go back 10 seconds')}
-          />
+function ReplayCurrentTime() {
+  const {currentTime, duration} = useReplayContext();
+
+  return (
+    <span>
+      {formatTime(currentTime)} / {duration ? formatTime(duration) : '??:??'}
+    </span>
+  );
+}
+
+function ReplayPlaybackSpeed({speedOptions}: {speedOptions: number[]}) {
+  const {setSpeed, speed} = useReplayContext();
+
+  return (
+    <ButtonBar active={String(speed)} merged>
+      {speedOptions.map(opt => (
+        <Button
+          key={opt}
+          size="xsmall"
+          barId={String(opt)}
+          onClick={() => setSpeed(opt)}
+          title={t('Set playback speed to %s', `${opt}x`)}
+        >
+          {opt}x
+        </Button>
+      ))}
+    </ButtonBar>
+  );
+}
+
+const ReplayControls = ({
```

Comment:
I can't tell, did more stuff get added in here?

Reply:
Nothing added. The internals of the component delegate more to other smaller components, and the <Panel> stuff isn't separate, so there's no more React.Fragment.

I've reading this PR using Split view, with "hide whitespace" enabled and I think it makes this file in particular easier to grok .